File: pushbullet/pushbullet_lambda.py
import json
import os
import requests

def lambda_handler(event, context):
    sns = event['Records'][0]['Sns']
    push_title = sns['Subject']
    push_body = sns['Message']

    data = json.dumps({'title': push_title, 'body': push_body, 'type': 'note'})

    pushbullet_access_token = os.environ.get('PUSHBULLET_ACCESS_TOKEN')
    
    response = requests.post('https://api.pushbullet.com/v2/pushes', \
        headers={'Access-Token': pushbullet_access_token, 'Content-Type': 'application/json'}, \
        data=data
    )

    if response.status_code == 200:
        print('Successfully sent push')
    else:
        print('Push failed due to HTTP ' + str(response.status_code))

    return {
        'statusCode': response.status_code,
        'body': json.dumps(push_title+': '+push_body)
    }

File: pushbullet/pushbullet_lambda.go
package main

import (
    "context"
    "net/http"
    "encoding/json"
    "strings"
    "os"
    "github.com/aws/aws-lambda-go/lambda"
    "github.com/aws/aws-lambda-go/events"
)

type Notification struct {
    Type string `json:"type"`
    Title string `json:"title"`
    Body string `json:"body"`
}

func handler(ctx context.Context, snsEvent events.SNSEvent) {
    for _, record := range snsEvent.Records {
        snsRecord := record.SNS
        
        notification := &Notification{
            Title: snsRecord.Subject,
            Body:  snsRecord.Message,
            Type: "note",
        }
        
        push(*notification)
    }
}

func main() {
    lambda.Start(handler)
}

func push(notification Notification) int {
    access_token := os.Getenv("PUSHBULLET_ACCESS_TOKEN")

    body := to_json(notification)
    req, _ := http.NewRequest("POST", "https://api.pushbullet.com/v2/pushes", strings.NewReader(body))
    req.Header.Add("Access-Token", access_token)
    req.Header.Add("Content-Type", "application/json")

    client := &http.Client{}
    resp, _ := client.Do(req)
    defer resp.Body.Close()

    return resp.StatusCode
}

func to_json(data interface{}) string {
    jsonStr, _ := json.Marshal(data)
    return string(jsonStr)
}